java-agent/src/nativeMain/kotlin/com/epam/drill/agent/configuration/Configuration.kt

```diff
@@ -161,55 +189,57 @@ private fun addWsSchema(address: String?): String? {
-    logger.info { "config parameters: $configParams" }
-    val resultParams = configParams?.toMutableMap()?.also { it.putAll(agentParams) } ?: agentParams
-    logger.info { "result parameters: $resultParams" }
-    return resultParams
-}
 
 private fun readFile(filepath: String): String {
-    val isFilename: (String) -> Boolean = { Regex("[\\w\\-.]+").matches(it) }
-    var fileDescriptor = open(filepath, O_RDONLY)
-    if (fileDescriptor == -1 && isFilename(filepath))
-        fileDescriptor = open(drillInstallationDir + pathSeparator + filepath, EROFS)
-    if (fileDescriptor == -1)
-        throw IllegalArgumentException("Cannot open the config file with path=$drillInstallationDir, file=$filepath")
-    return Input(fileDescriptor).readText().also { close(fileDescriptor) }
+    val fileDescriptor = open(filepath, O_RDONLY)
+    return fileDescriptor
+        .takeIf { it != -1 }
+        ?.let { Input(it).readText().also { close(fileDescriptor) } }
+        ?: "".also { logger.error { "Cannot open the config file with path=$filepath" } }
 }
 
-private fun asAgentParams(
+private fun parseArguments(
     input: String?,
-    lineDelimiter: String = ",",
     filterPrefix: String = "",
-    mapDelimiter: String = "="
+    mapDelimiter: String = "=",
+    lineDelimiters: Array<String> = arrayOf(",")
 ): Map<String, String> {
     if (input.isNullOrEmpty()) return emptyMap()
     try {
-        return input.split(lineDelimiter)
+        return input.split(*lineDelimiters)
             .filter { it.isNotEmpty() && (filterPrefix.isEmpty() || !it.startsWith(filterPrefix)) }
             .associate { it.substringBefore(mapDelimiter) to it.substringAfter(mapDelimiter, "").trim() }
     } catch (parseException: Exception) {
         throw IllegalArgumentException("wrong agent parameters: $input")
     }
 }
 
-private fun parseAsAgentArguments(map: Map<String, String>) = AgentArguments::class.serializer().run {
-    val module = serializersModuleOf(this)
-    this.deserialize(SimpleMapDecoder(module, map))
+private fun getConfigFileParams(configFilePath: String): Map<String, String> {
+    return configFilePath
+        .runCatching(::readFile)
+        .getOrNull()
+        .let { parseArguments(it, filterPrefix = "#", lineDelimiters = arrayOf("\n\r", "\r\n", "\n", "\r")) }
+}
+
+private fun getConfigFilePath(
+    commandLineParams: Map<String, String>,
+    drillInstallationDir: String
+) = (commandLineParams[CONFIG_PATH_PARAM]
+    ?: getenv(SYSTEM_CONFIG_PATH)?.toKString()
+    ?: "${drillInstallationDir}${pathSeparator}drill.properties")
+
+private fun getDrillInstallationDir(commandLineParams: Map<String, String>): String {
+    return commandLineParams[DRILL_INSTALLATION_DIR_PARAM]
+        ?: getAgentPathCommand()?.let { parseAgentDirFromAgentPathCommand(it) }
+        ?: "."
+}
+
+private fun getAgentPathCommand(): String? {
+    return getenv("JAVA_TOOL_OPTIONS")?.toKString() ?: getAgentPathCommandFromProcess()
```

Comment:
inline at callsight

Reply:
`getAgentPathCommandFromProcess()` might throw - we must at least rethrow with clear error description